include/nighthawk/client/process_context.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "envoy/network/address.h"
+#include "envoy/stats/store.h"
+
+#include "nighthawk/client/client_worker.h"
+#include "nighthawk/client/factories.h"
+#include "nighthawk/client/options.h"
+#include "nighthawk/client/output_formatter.h"
+#include "nighthawk/common/statistic.h"
+#include "nighthawk/common/uri.h"
+
+#include "common/api/api_impl.h"
+#include "common/common/logger.h"
+#include "common/event/real_time_system.h"
+
+namespace Nighthawk {
+namespace Client {
+
+/**
+ * Process context is shared between the CLI and grpc service. It is capable of executing
+ * a full Nighthawk test run. Only a single instance is allowed at a time.
+ */
+class ProcessContext {
```

Comment:
I take it this isn't related to Envoy's recently introduced `ProcessContext`?

Reply:
Unrelated indeed, it's a coincidence

Reply:
How come `ProcessContext` and not `Process`? Don't want to bike shed, just trying to understand if this is actually a context or an execution wrapper.

Reply:
Renamed to `Process` as you are right in that it actually is an execution wrapper